Langgraph integration

s-udhaya opened the pull request:
### Related Issues/PRs

<!-- Uncomment 'Resolve' if this PR can close the linked items. -->
<!-- Resolve --> None

### What changes are proposed in this pull request?

Implement function calling agent using langgraph. Tested with UC tool and vector search retriever tool

### How is this PR tested?
<!-- Manual testing is currently required to merge pull requests. Please include a screenshot/video proof of manual testing of changes
 -->
<img width="1912" alt="Screenshot 2024-12-06 at 15 43 42" src="https://github.com/user-attachments/assets/cfb7d973-17a6-4726-915d-e1829538a67c">

FMurray commented on langgraph_agent_app_sample_code/cookbook/agents/multi_agent_supervisor.py:
@s-udhaya can the supervisor be a langgraph model? I think in practice this is how most users with LangGraph code will expect to integrate. 

@smurching @niall-turbitt wdyt?

s-udhaya replied:
@FMurray Multiagent implementation for sure can be a langgraph model. This one is a direct copy from the opensdk implementation and multiagent was not part of the initial implementation. I am now in the process of cleaning this up. Will convert this PR to draft.

s-udhaya commented:
@FMurray I have removed the unused modules and simplified the configurations. While I'm working on the multi-agent functionality, could you please do a quick review of the PR? Thanks!
